core/src/main/scala/sttp/tapir/Codec.scala

```diff
@@ -199,59 +199,69 @@ object Codec extends CodecExtensions with FormCodecMacros with CodecMacros with
   def rawPart(
       partCodecs: Map[String, PartCodec[_, _]],
       defaultCodec: Option[PartCodec[_, _]]
-  ): Codec[Seq[RawPart], Seq[AnyPart], MultipartFormData] =
-    new Codec[Seq[RawPart], Seq[AnyPart], MultipartFormData] {
+  ): Codec[Seq[RawPart], Map[String, _], MultipartFormData] =
```

Comment:
Unfortunately we lost some flexibility here, as it's now not possible to encode a sequence of parts into raw parts (something that was the main purpose of the `rawPart` codec before). So somehow we need to regain that functionality.

There's also the problem that a `Map` looses ordering, and I think we should preserve the order of parts. So summing up, we need a codec which encodes an ordered sequence of parts into raw parts.

How can we achieve this ... some ideas:

(1) `Codec[Seq[RawPart], ListMap[Part[Unit], _], MultipartFormData]` - the map keys would be part meta-data (with the body always a unit), and the values bodies.

(2) keep the old `Codec[Seq[RawPart], Seq[AnyPart], MultipartFormData]`, and require `partCodecs: Map[String, PartCodec[_, Part[_]]`, that is that each codec encodes to a part. This is somewhat constraining, but maybe we could make it work with appropriate implicit conversions like `partToPart` and `partBasic` below so that it's usable with normal codecs

Reply:
I was assuming that users can either use `.multipartBody` which produces `Seq[Part[Array[Byte]]]` or `.multipartBody[T]` where `T` is some case class. What functionality is missing exactly?

Reply:
deserialising into a sequence of `Part`s, with codecs for some (or all) parts specified. E.g. I want the `file` and `photo` parts to be decoded into files, and the rest read into memory as byte arrays

Reply:
then you can still define your case class as
```scala
case class Upload(
  file: Part[File],
  photo: Part[File],
  somethingElse1: Part[Array[Byte]],
  somethingElse2: Part[Array[Byte]]
)
```

Reply:
yeah although this does require you to define a case class - I was talking about something more basic, on top of which you can create codecs which read data into a case class - that is, when you just want the parts as a sequence in the order in which they were defined originally, where you specify how individual parts should be handled on a low level

the `Map[String, _]` is I think unnecessarily constrained

Reply:
Maybe we can have two methods? The old one `Codec.multipart` that does what it previously did (creates `MultipartCodec[Seq[AnyPart]]`) and the new one, let's call it `Codec.multipartMap` producing `MultipartCodec[Map[String, _]]` used for the sole purpose of using it to create a case class instance?

Reply:
But a `MultipartCodec[List[AnyPart]]` is strictly more powerful than `MultipartCodec[Map[String, _]`, and the second can be easily derived from the first. You can also use it to derive any kind of codec for a case class instance (as you have the `Part` metadata)